woocommerce-gateway-stripe.php

```diff
@@ -154,6 +154,7 @@ public function init() {
 				require_once dirname( __FILE__ ) . '/includes/compat/class-wc-stripe-pre-orders-compat.php';
 				require_once dirname( __FILE__ ) . '/includes/class-wc-gateway-stripe.php';
 				require_once dirname( __FILE__ ) . '/includes/class-wc-stripe-feature-flags.php';
+				require_once dirname( __FILE__ ) . '/includes/notes/class-wc-stripe-upe-availability-note.php';
```

Comment:
Instead of loading the Note class and then checks `trait_exists()` inside, what if we do something similar to WCPay where we check it here? https://github.com/Automattic/woocommerce-payments/blob/d4e86120cacdc077cd4858ea7e9a9071358b6630/includes/class-wc-payments.php#L878-L893

```
if (version > 4.4 ) {
  require_once dirname( __FILE__ ) . '/includes/notes/class-wc-stripe-upe-availability-note.php';
}
```

Would this help preventing the entire note to load without adding all the `if ( class_exists( 'WC_Stripe_UPE_Availability_Note',` everywhere?

Reply:
This actually solved the issue. I changed the implementation mention in [this](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1762#discussion_r702152140) and [this](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1762#pullrequestreview-746460458) comments also after doing this and everything seems to be in place now.
Thanks for pointing out in the right direction :grinning: